mrp_production_auto_validate/models/mrp_production.py

```diff
@@ -39,11 +42,12 @@ def check_bom_auto_validate(self):
                     ).format(qty=mo.bom_id.product_qty)
                 )
 
-    @api.onchange("bom_id")
-    def _onchange_bom_id(self):
-        res = super()._onchange_bom_id()
-        self.auto_validate = self.bom_id.mo_auto_validation
-        return res
+    @api.depends("bom_id.mo_auto_validation", "state")
+    def _compute_auto_validate(self):
+        for prod in self:
+            if prod.state != "draft":
+                continue
```

Comment:
```suggestion
                prod.auto_validate = False
                continue
```

Reply:
No, that would mean `auto_validate` is set to `False` as soon as the MO is confirmed.

Goal with this is to avoid recomputing the value once the MO is confirmed. e.g. if the value changes on the BOM but the MO was already confirmed, or if the user forces another value while the MO is in draft, we don't want to change the value after confirmation.

I agree it looks kind of weird to not have a value being assigned, but as a MO is always supposed to be created in draft, the value will always be initialized.

I added this as a comment for explanation.